src/operations/count_documents.ts

```diff
@@ -13,7 +13,7 @@ export interface CountDocumentsOptions extends AggregateOptions {
 }
 
 /** @internal */
-export class CountDocumentsOperation extends AggregateOperation<number> {
+export class CountDocumentsOperation extends AggregateOperation {
```

Comment:
`AggregateOperation` is not generic and if it were it can't be that a subclass returns an incompatible type from the parent. (ex. Parent -> A -> B, B cannot have a method incompatible with Parent otherwise things that expect "Parent" will break when given Bs) This and the command below are `TODO_NODE_3286` related. The operations layer TS does not work well and here in particular because the expectation (and reality) is that every command returns an object, I think returning a "number" should be handled up in the `coll.countDocument` API. Taking it even further I don't see the value in this subclass? can we just call `this.aggregate` from within `collection.countDocuments`?

Stuck the appropriate type annotation on here.

Reply:
Yeah, agreed - the preferable approach would be to remove this class entirely (up to you).  We do have this precedent with our index management APIs too.

Reply:
https://github.com/mongodb/node-mongodb-native/pull/4138

Reply:
I think it would make sense to have that change be a distinct change from what is in this PR so it is isolated in the history and revertable on its own. Can we resolve this?